tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java

```diff
@@ -174,29 +187,74 @@ public String getName() {
 
         @Override
         public boolean is(String item) {
+            if (item.equals("root")) {
+                item = "und";
+            }
             if (!item.contains("_")) {
-                return lang.is(item);
+                return checkLang(item);
             }
             LanguageTagParser ltp;
             try {
                 ltp = new LanguageTagParser().set(item);
             } catch (Exception e) {
                 return false;
             }
-            return lang.is(ltp.getLanguage())
+            return checkLang(ltp.getLanguage())
                     && (ltp.getScript().isEmpty() || script.is(ltp.getScript()))
                     && (ltp.getRegion().isEmpty() || region.is(ltp.getRegion()))
                     && (ltp.getVariants().isEmpty() || and(variant, ltp.getVariants()))
                     && ltp.getExtensions().isEmpty()
                     && ltp.getLocaleExtensions().isEmpty();
         }
 
+        public boolean checkLang(String language) {
+            return lang.is(language);
+        }
+
         @Override
         public String getSample() {
             return "de";
         }
     }
 
+    /**
+     * Check for the language OR certain backwards-compatible exceptions for data to support
+     * retaining variants, namely likelySubtags: "in","iw","ji","jw","mo","tl"
+     */
+    public static class XLocaleMatchValue extends LocaleMatchValue {
```

Comment:
personal style nit (you can choose your own style): have the explanation in the name
```suggestion
    public static class LocaleOrDeprecatedLocaleMatchValue extends LocaleMatchValue {
```

Reply:
Thanks!